tls/s2n_resume.c

```diff
@@ -304,6 +304,8 @@ static S2N_RESULT s2n_deserialize_resumption_state(struct s2n_connection *conn,
             RESULT_GUARD(s2n_tls12_client_deserialize_session_state(conn, from));
         }  
     } else if (format == S2N_TLS13_SERIALIZED_FORMAT_VERSION) {
+        /* Stateful session resumption is not yet impelemented in TLS1.3 */
+        RESULT_ENSURE_EQ(conn->config->use_tickets, 1);
```

Comment:
Is use_tickets not true when doing stateful session resumption?

(Also, nit: s/impelemented/implemented)

Reply:
`config->use_tickets ` is enabled only for stateless session resumption when `s2n_config_set_session_tickets_onoff` is called. For stateful session resumption we have `config->use_session_cache`.

Reply:
Are you sure? Where is config->use_session_cache used?

https://github.com/aws/s2n-tls/blob/main/tls/s2n_handshake_io.c#L808-L826

Reply:
From offline discussion: 
`config->use_tickets` does not determine if the connection is stateful or stateless, it only indicates we should issue/send tickets. `config->use_session_cache` is enabled only for session resumption with session ID and can be used here.

Reply:
As the server s2n receives a `conn->session_id` for stateful session resumption when `s2n_connection_set_session` is called:  

https://github.com/aws/s2n-tls/blob/3d522c4b8f2071c2983d3a00269e3ffbb41ea4d5/tls/s2n_resume.c#L325

and `conn->client_ticket` for stateful session resumption when `s2n_connection_set_session` is called:  

https://github.com/aws/s2n-tls/blob/3d522c4b8f2071c2983d3a00269e3ffbb41ea4d5/tls/s2n_resume.c#L341-L342